src/App.tsx

```diff
@@ -1,35 +1,75 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import './App.css';
+import CarForm from './components/car/car_form';
+import { useState } from 'react';
+import CarList from './components/car/car_list';
+import { Car } from './types';
+import MaterialUISwitch from './components/materialUISwitch'; 
+import { createTheme, ThemeProvider } from '@mui/material';
+import {  useEffect } from 'react';
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [cars, setCars] = useState<Car[]>([]);
+  const [darkMode, setDarkMode] = useState<boolean>(false);
+
+  const addCar = (car: Car) => {
+    setCars([...cars, car]);
+  };
+
+  const toggleDarkMode = () => {
+    setDarkMode(!darkMode);
+  };
+
+  const theme = createTheme({
+    palette: {
+      mode: darkMode ? 'dark' : 'light',
+      primary: {
+        main: '#646bf3',
+      },
+      secondary: {
+        main: '#fff',
+      },
+      background: {
+        default: darkMode ? '#101113' : '#fff',
+      },
+      text: {
+        primary: darkMode ? '#fefefe' : '#101113',
+      },
+    },
+  });
+
+  useEffect(() => {
+    document.body.style.backgroundColor = theme.palette.background.default;
+    document.body.style.color = theme.palette.text.primary;
+  }, [theme]);
+  
 
   return (
-    <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
+    <ThemeProvider theme={theme}>
+      <div className='app-container'>
+      <div className="theme-toggle" style={{ position: 'fixed', top: '20px', right: '20px' }}>
+
+        <MaterialUISwitch
+          checked={darkMode}
+          onChange={toggleDarkMode}
+          inputProps={{ 'aria-label': 'toggle dark mode' }}
+          color="secondary"
+          style={{ marginRight: '10px' }}
+        />
       </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
+
+        <div className='car-form-container'>
+          <div className='car-form-add'>
+            <CarForm addCar={addCar} />
+          </div>
```

Comment:
Try to replace html `div` with some **MUI** containers, for example `<Box>`, `<Stack>` or `<Grid>`. It will also simplify the layout since you can use `sx` prop to configure it